heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -414,85 +414,76 @@ export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
-          const longestMatch: number = Math.max(...packageDirMatches.map((match) => match.length));
-          const restOfPath: string = path.normalize('./' + value.substr(longestMatch));
-          value = path.resolve(resolvedPackagePath, restOfPath);
+          // First entry is the entire match
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(packageDirMatches[0].length));
+          propertyValue = path.resolve(resolvedPackagePath, restOfPath);
         }
 
         // Return early, since the remainder of this function is used to resolve module paths
         if (!options.resolveAsModule) {
-          return value;
+          return propertyValue;
         }
 
         // Example:  @rushstack/heft-jest-plugin
-        if (value === PLUGIN_PACKAGE_NAME) {
-          return JestPlugin._ownPackageFolder;
+        if (propertyValue === PLUGIN_PACKAGE_NAME) {
+          return OWN_PACKAGE_FOLDER;
         }
 
         // Example:  @rushstack/heft-jest-plugin/path/to/file.js
-        if (value.startsWith(PLUGIN_PACKAGE_NAME)) {
-          const restOfPath: string = path.normalize('./' + value.substr(PLUGIN_PACKAGE_NAME.length));
-          return path.join(JestPlugin._ownPackageFolder, restOfPath);
-        }
-
-        // Attempt to resolve the specified module as we would using PathResolution.NodeResolve,
-        // using a prefix if one was provided.
-        try {
-          return Import.resolveModule({
-            baseFolderPath: configurationFileDir,
-            modulePath: `${options.modulePrefix || ''}${value}`
-          });
-        } catch (e) {
-          // Rethrow if the no prefix was provided and we should not ignore missing modules
-          if (!options.modulePrefix && !options.ignoreMissingModule) {
-            throw e;
-          }
-        }
-
-        // Finally, attempt to resolve the specified module without a prefix as we would using
-        // PathResolution.NodeResolve. Only throw if we shouldn't ignore missing modules. This
-        // option is generally provided to allow Jest to resolve modules that are sourced from
-        // Jest itself, ex. 'jest-environment-jsdom' is resolved from Jest by default
-        // See: https://github.com/facebook/jest/blob/0a902e10e0a5550b114340b87bd31764a7638729/packages/jest-config/src/utils.ts#L178
-        try {
-          return Import.resolveModule({
-            baseFolderPath: configurationFileDir,
-            modulePath: value
-          });
-        } catch (e) {
-          if (!options.ignoreMissingModule) {
-            throw e;
-          }
+        if (propertyValue.startsWith(PLUGIN_PACKAGE_NAME)) {
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(PLUGIN_PACKAGE_NAME.length));
+          return path.join(OWN_PACKAGE_FOLDER, restOfPath);
         }
 
-        return value;
+        return !options.modulePrefix
+          ? jestResolve(/*resolver:*/ undefined, {
```

Comment:
What is this first argument for normally?

Reply:
Normally it's for the resolver loaded through the `resolver` setting on the Jest configuration. We _could_ add support for it, however it would require some tweaks to the `heft-config-file` package to provide us the current state of the configuration file/the unresolved current configuration file + the resolved state of the combined configuration file up to this point.

Reply:
Circling back to https://github.com/microsoft/rushstack/pull/2761 :

Suppose that I specify `"testEnvirionment": "shpadoinkle"` which Jest would normally resolve as `jest-environment-shpadoinkle`.

How does `jest-environment-shpadoinkle` get resolved now?  Can it be provided by the rig?  Can it be overridden by my local project's **package.json**?

Reply:
It can be provided by rig packages. Currently it would be resolved in this manner for `testEnvironment` (or other prefixed values like `jest-runner-`):
- resolve the prefixed package name (ex. `jsdom` -> `jest-environment-jsdom`) relative to the directory the configuration file is specified in using Node resolution
- resolve the prefixed package name from `jest-config` using `require.resolve`
- resolve the unprefixed package name relative to the directory the configuration file is specified in using Node resolution
- resolve the unprefixed package name from `jest-config` using `require.resolve`